### PR TITLE
mupdf: update to 1.23.4.

### PR DESCRIPTION
**Pull request by mhmdanas**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by dkwo** [comment]
do fbpdf, sioyek, zathura work with this?

**Comment by mhmdanas** [comment]
Both sioyek and zathura work, didn't test fbpdf.

**Comment by Zapeth** [comment]
It looks like sioyek needs to be patched for mupdf 1.23 -> https://github.com/ahrm/sioyek/pull/831 (just saw the PR, haven't tested it)

File: srcpkgs/mupdf/template [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Template file for 'mupdf'
+# Static library, revbump all dependants on mupdf updates
```
Comment (mhmdanas) [inline srcpkgs/mupdf/template:2]: We could do what Alpine does maybe: https://git.alpinelinux.org/aports/tree/community/mupdf/APKBUILD#n109

But not sure if it's worth it.

**Comment by dkwo** [comment]
the commit msg for sioyek needs changed. also, the patch is not merged upstream. have you tested it?
how urgent is the mupdf update? what does it fix?

**Comment by mhmdanas** [comment]
> the commit msg for sioyek needs changed.

I don't think so, just because it adds a patch doesn't mean it has to have a different commit message. But I'll change it anyway, if you insist. :)

> also, the patch is not merged upstream. have you tested it?

I have, it seems to be working fine.

> how urgent is the mupdf update? what does it fix?

It's not urgent as far as I know. You can see [the changelog](https://github.com/ArtifexSoftware/mupdf/blob/master/CHANGES) for the list of changes and fixes.

**Comment by benjcal** [comment]
just a quick comment that this broke sioyek for me, but so far for only one PDF

https://ww1.microchip.com/downloads/aemDocuments/documents/MCU08/ProductDocuments/DataSheets/AVR64DD-14-20-Prelim-DataSheet-DS40002444.pdf

I'm able to open it and scroll up and down, but the moment that I open the table of contents and go anywhere sioyek just shows a blank screen.

Reverting to `sioyek-2.0.0_4` solved the issue with the PDF in question.

:-)

**Comment by Zapeth** [comment]
It sounds like this issue in sioyek https://github.com/ahrm/sioyek/issues/539#issuecomment-1758664514 which seems like its resolved upstream

You could try to compile it with the fix and PR the patch if it works (or wait for the next release)

**Pull request merged**